gitea/giteaclient.go

```diff
@@ -0,0 +1,371 @@
+	// DeleteKey is a wrapper for "DELETE /repos/{owner}/{repo}/keys/{key_id}".
+	// This function handles HTTP error wrapping.
+	DeleteKey(ctx context.Context, owner, repo string, id int64) error
+
+	// GetTeamPermissions is a wrapper for "GET /repos/{owner}/{repo}/teams/{team_slug}
+	// This function handles HTTP error wrapping, and validates the server result.
+	GetTeamPermissions(ctx context.Context, orgName, repo, teamName string) (*gitea.AccessMode, error)
+	// ListRepoTeams is a wrapper for "GET /repos/{owner}/{repo}/teams".
+	// This function handles pagination, HTTP error wrapping, and validates the server result.
+	ListRepoTeams(ctx context.Context, orgName, repo string) ([]*gitea.Team, error)
+	// AddTeam is a wrapper for "PUT /repos/{owner}/{repo}/teams/{team_slug}".
+	// This function handles HTTP error wrapping.
+	AddTeam(ctx context.Context, orgName, repo, teamName string, permission gitprovider.RepositoryPermission) error
+	// RemoveTeam is a wrapper for "DELETE /repos/{owner}/{repo}/teams/{team_slug}".
+	// This function handles HTTP error wrapping.
+	RemoveTeam(ctx context.Context, orgName, repo, teamName string) error
+}
+
+type giteaClientImpl struct {
+	c                  *gitea.Client
+	destructiveActions bool
+}
+
+var _ giteaClient = &giteaClientImpl{}
+
+func (c *giteaClientImpl) Client() *gitea.Client {
+	return c.c
+}
+
+func (c *giteaClientImpl) GetOrg(_ context.Context, orgName string) (*gitea.Organization, error) {
+	apiObj, res, err := c.c.GetOrg(orgName)
+	if err != nil {
+		return nil, handleHTTPError(res, err)
+	}
+	// Validate the API object
+	if err := validateOrganizationAPI(apiObj); err != nil {
+		return nil, err
+	}
+	return apiObj, nil
+}
+
+func (c *giteaClientImpl) ListOrgs(_ context.Context) ([]*gitea.Organization, error) {
+	opts := gitea.ListOrgsOptions{}
+	apiObjs := []*gitea.Organization{}
+	listOpts := &opts.ListOptions
+
+	err := allPages(listOpts, func() (*gitea.Response, error) {
+		// GET /user/orgs"
+		pageObjs, resp, listErr := c.c.ListMyOrgs(opts)
+		if len(pageObjs) > 0 {
+			apiObjs = append(apiObjs, pageObjs...)
+			return resp, listErr
+		}
+		return nil, nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// Validate the API objects
+	for _, apiObj := range apiObjs {
+		if err := validateOrganizationAPI(apiObj); err != nil {
+			return nil, err
+		}
+	}
+	return apiObjs, nil
+}
+
+func (c *giteaClientImpl) ListOrgTeamMembers(ctx context.Context, orgName, teamName string) ([]*gitea.User, error) {
+	teams, err := c.ListOrgTeams(ctx, orgName)
+	if err != nil {
+		return nil, err
+	}
+	apiObjs := []*gitea.User{}
+	teamFound := false
+	for _, team := range teams {
+		if team.Name != teamName {
+			continue
+		}
+		teamFound = true
+		users, _, err := c.c.ListTeamMembers(team.ID, gitea.ListTeamMembersOptions{})
+		if err != nil {
+			continue
+		}
+		apiObjs = append(apiObjs, users...)
+	}
+	if teamFound {
+		return apiObjs, nil
+	}
+	return nil, gitprovider.ErrNotFound
```

Comment:
As we expect only one team with the same, we should exit early after retrieving the team. We could do something like:

```go
for _, team := range teams {
  if team.Name == teamName {
    users, _, err := c.c.ListTeamMembers(team.ID, gitea.ListTeamMembersOptions{})
    if err != nil {
	return nil, fmt.Errorf("failed to list team %s members: %w", teamName, err)
    }
    return users, nil
}
return nil, gitprovider.ErrNotFound
```